rust/feature-flags/src/flag_matching.rs

```diff
@@ -732,12 +734,29 @@ impl FeatureFlagMatcher {
                     .await;
             }
 
-            // NB: we can only evaluate group or person properties, not both
+            // Separate cohort and non-cohort filters
+            let (cohort_filters, non_cohort_filters): (Vec<PropertyFilter>, Vec<PropertyFilter>) =
+                flag_property_filters
+                    .iter()
+                    .cloned()
+                    .partition(|prop| prop.prop_type == "cohort");
```

Comment:
I want to be able to evaluate the cohort properties and non-cohort properties with different methods, since I have to handle the `in`/`not_in` case for cohorts that I don't have to deal with using `match_properties`.